cmd/meroxa/root/apps/deploy.go

```diff
@@ -649,25 +638,54 @@ func (d *Deploy) Execute(ctx context.Context) error {
 		return err
 	}
 
-	if err := d.validateSpecVersionDeployment(); err != nil {
+	gitSha, err := d.turbineCLI.GetGitSha(ctx)
+	if err != nil {
 		return err
 	}
 
-	// ⚠️ This is only until we re-deploy applications applying only the changes made
-	if err := d.tearDownExistingResources(ctx); err != nil {
-		return err
+	var app *meroxa.Application
+	if d.specVersion == "" {
+		// ⚠️ This is only until we re-deploy applications applying only the changes made
+		if err := d.tearDownExistingResources(ctx); err != nil {
+			return err
+		}
+	} else {
+		// Intermediate Representation Workflow
+		if err := d.validateSpecVersionDeployment(); err != nil {
+			return err
+		}
+
+		app, err = d.client.CreateApplication(ctx, &meroxa.CreateApplicationInput{
+			Name:     d.appName,
+			Language: d.lang,
+			GitSha:   gitSha,
+			Pipeline: meroxa.EntityIdentifier{}})
+		if err != nil && !strings.Contains(err.Error(), "already exists") {
+			return err
+		}
 	}
 
-	err := d.prepareDeployment(ctx)
+	err = d.prepareDeployment(ctx)
 	defer d.turbineCLI.CleanUpBinaries(ctx, d.appName)
 	if err != nil {
 		return err
 	}
 
-	gitSha, err := d.turbineCLI.GetGitSha(ctx)
+	err = d.deployApp(ctx, d.fnName, gitSha, d.specVersion)
 	if err != nil {
 		return err
 	}
 
-	return d.deployApp(ctx, d.fnName, gitSha, d.specVersion)
+	if d.specVersion == "" {
+		app, err = d.checkPreIRApplication(ctx)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
@TODO get the latest deployment status, potentially in a loop if it's not implemented as a stream